src/theory/arith/arith_rewriter.cpp

```diff
@@ -188,6 +188,91 @@ RewriteResponse ArithRewriter::postRewriteAtom(TNode atom)
   rewriter::addToSum(sum, left, negate);
   rewriter::addToSum(sum, right, !negate);
 
+  if (kind != Kind::EQUAL)
```

Comment:
I suggest moving this to a separate function, or adding a comment that we are now handling the case of bv2nat, so that it is clear that this part is separated from the rest.